staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -210,13 +212,13 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			http.Error(w, fmt.Sprintf("%v not allowed while CustomResourceDefinition is terminating", requestInfo.Verb), http.StatusMethodNotAllowed)
 			return
 		}
-		handler = handlers.CreateResource(storage, requestScope, discovery.NewUnstructuredObjectTyper(nil), r.admission)
+		handler = handlers.CreateResource(storage, crdInfo.requestScopes[crdInfo.storageVersion], discovery.NewUnstructuredObjectTyper(nil), r.admission)
```

Comment:
having the storageVersion scope surprises me here

Reply:
I've moved the logic into storage strategy. Now we have an storage object per version but the strategy's prepare for create and update will call the converters to convert them into storage version.